src/Analyzer/SetUtils.cpp

```diff
@@ -147,20 +159,34 @@ Block getSetElementsForConstantValue(const DataTypePtr & expression_type, const
     if (lhs_type_depth == rhs_type_depth)
     {
         /// 1 in 1; (1, 2) in (1, 2); identity(tuple(tuple(tuple(1)))) in tuple(tuple(tuple(1))); etc.
-
         Array array{value};
-        result_block = createBlockFromCollection(array, set_element_types, transform_null_in);
+        DataTypes value_types{value_type};
+        result_block = createBlockFromCollection(array, value_types, set_element_types, transform_null_in);
     }
     else if (lhs_type_depth + 1 == rhs_type_depth)
     {
         /// 1 in (1, 2); (1, 2) in ((1, 2), (3, 4))
-
         WhichDataType rhs_which_type(value_type);
 
         if (rhs_which_type.isArray())
-            result_block = createBlockFromCollection(value.get<const Array &>(), set_element_types, transform_null_in);
+        {
+            const DataTypeArray * value_array_type = typeid_cast<const DataTypeArray *>(value_type.get());
+            size_t value_array_size = value.get<const Array &>().size();
+            DataTypes value_types;
+            value_types.reserve(value_array_size);
+
+            for (size_t i = 0; i < value_array_size; ++i)
+            {
+                value_types.push_back(value_array_type->getNestedType());
+            }
```

Comment:
Why not just `DataTypes value_types(value_array_size, value_array_type->getNestedType())`?

Reply:
done